nifi-assembly/pom.xml

```diff
@@ -350,6 +350,42 @@ language governing permissions and limitations under the License. -->
             <version>2.0.0-SNAPSHOT</version>
             <type>nar</type>
         </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-generic-db-adapter-provider-nar</artifactId>
+            <version>2.0.0-SNAPSHOT</version>
+            <type>nar</type>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-oracle-db-adapter-provider-nar</artifactId>
```

Comment:
Is the word 'provider' helpful in these?  

It seems to me we have database adapter api and database adapters that implement the api.  Ie we have

nifi-database-adapter-api is the api module.

nifi-oracle-db-adapter-nar implements the api specific to oracle column/types.

Provider seems not needed.  Can you share more if that is helpful?

Reply:
I'm fine with removing the word 'provider' from the controller services, but I'm not sure if it's a good practice, since we have actual 'adapter' classes, that are instantiated by the 'provider' classes. If you feel like you have better wording for this concept, then I'm OK with renaming.

Or did you mean this 'adapter' - 'provider' abstraction does not seem useful? IMHO it adds some flexibility to how we organize, or combine various adapters, because for example a single controller service for MSSQL can provide adapters for multiple versions of that dialect.

Reply:
I dont think I realized we had a 'database-adapter-api' and a 'database-adapter-provider-api'. 

The word 'adapter' is generic/broad as is 'provider' so it just looks odd having them both side by side.

I dont know enough about why we need these two separate APIs so I guess I'll back out of this comment for now.